barman/config.py

```diff
@@ -1676,6 +1687,317 @@ def _validate_with_keys(config_items, allowed_keys, section):
+        old_file_name = self.queue_file + ".old"
+        try:
+            os.rename(self.queue_file, old_file_name)
+        except FileNotFoundError:
+            old_file_name = None
+        os.rename(self.queue_file + ".tmp", self.queue_file)
+        if old_file_name:
+            os.remove(old_file_name)
+        self._queue = None
+
+
+class ConfigChangesProcessor:
+    """
+    The class is responsible for processing the config changes to
+    apply to the barman config
+    """
+
+    def __init__(self, config):
+        self.config = config
+        self.applied_changes = []
+
+    def receive_config_changes(self, changes):
+        """
+        Process all the configuration changes
+        """
+        # Get all the available configuration change files in order
+        changes_list = []
+        for section in changes:
+            section_name = section["server_name"]
+            # IMPORTANT: i'm not yet entirely sure why but if we don't specify
+            # that changes_set is an empty list, a reference to a list to all
+            # the previous changes is passed. needs more investigation but this
+            # fixed the issue for now
+            chg_set = ConfigChangeSet(section=section_name, changes_set=[])
+            for json_cng in section:
+                if json_cng in ("server_name", "scope"):
+                    continue
+                file_name = self.config._config.get_config_source(
+                    section_name, json_cng
+                )
+                # if the configuration change overrides a default value
+                # then the source file is ".barman.auto.conf"
+                if file_name == "default":
+                    file_name = os.path.expanduser(
+                        "%s/.barman.auto.conf" % self.config.barman_home
+                    )
+                chg = None
+                # Instantiate the configuration change object
+                chg = ConfigChange(
+                    json_cng,
+                    section[json_cng],
+                    file_name,
+                )
+                chg_set.changes_set.append(chg)
+            changes_list.append(chg_set)
+
+        # If there are no configuration change we've nothing to do here
+        if len(changes_list) == 0:
+            _logger.debug("No valid changes submitted")
+            return
+
+        # Extend the queue with the new changes
+        with ConfigChangesQueue(self.config.config_changes_queue) as changes_queue:
+            changes_queue.queue.extend(changes_list)
+
+    def process_conf_changes_queue(self):
+        """
+        Process the configuration changes in the queue.
+
+        This method iterates over the configuration changes in the queue and applies them one by one.
+        If an error occurs while applying a change, it logs the error and raises an exception.
+
+        Raises:
+            Exception: If an error occurs while applying a change.
```

Comment:
```suggestion
        :raises:
            :exc:`Exception`: If an error occurs while applying a change.
```

Same comment about reST vs Google.